### PR TITLE
Feature/enable development files installation

### PR DESCRIPTION
**Pull request by RafaelPalomar**
Hello. Here some patches to make development files installable.

### CODE REVIEW

**Comment by pieper** [comment]
Hi - What's the context for wanting the developer files?  Does this also package them?  (i.e. make them part of the installer?)

**Comment by jcfr** [comment]
Thanks for your contribution :pray: 

In the coming days, I will carefully review your changes and start to selectively integrate them. If any, I will post questions here.

**Comment by lassoan** [comment]
Will this allow building C++ loadable modules without building Slicer?

**Comment by RafaelPalomar** [comment]
Hi. Thank all you for your quick reply.

@pieper this is in the context of getting the development files installed along the install tree and develops on the use of the Slicer_INSTALL_DEVELOPMENT flag already existing in 3D Slicer. This is useful if one distributes 3D Slicer as an installable binary package that developers could use (e.g., creating new modules). In Debian-based distributions, for instance, this applies to packages ending in "-dev" which contain files needed for development.

@pieper These patches uses CMake infrastructure similar to that used for installing libraries and applications. I would think that, at least, these patches gets us closer to that functionality that you mentioned, but I have not tried since I'm not very familiar with CPack. At least, I built 3D Slicer with Slicer_SUPERBUILD=OFF and Slicer_INSTALL_DEVELOPMENT=ON. After `make install` I could build loadable modules separately, based on the install tree.

@jcfr  I see the tests are not passed (that was unexpected). I will try to fix it before you do the integration. I'll message you when this is done.

@lassoan You have to build Slicer, or at least have the binaries distributed somehow, e.g., install tree in the system directories, since you need to link to some 3D Slicer libraries.

**Comment by pieper** [comment]
Hi @RafaelPalomar - thanks for the extra context.  It's a good goal and it would really help developers if we could provide a -dev version of Slicer for easy installation and testing.  We have always struggled to integrate Slicer's build with standard packaging for linux because we often find that we need to patch our dependencies (e.g. have a customized version of VTK or ITK with a few fixes).

**Comment by RafaelPalomar** [comment]
Hi @jcfr @pieper @lassoan. Here just a reminder as suggested by @jcfr.
I'm planning to join the Project Week in January. If you have the time, we could have a chat and I could show you what I'm up to with these changes.

**Comment by pieper** [comment]
Yes @RafaelPalomar, a chat a project week is perfect.  Please an entry to the project list under Infrastructure so we can flesh out ideas on a project page and link to this PR. https://projectweek.na-mic.org/PW33_2020_GranCanaria/

File: CMake/UseSlicer.cmake.in [hunk 1]
```diff
@@ -310,9 +316,9 @@ endif()
 # Set Slicer_INSTALL_*MODULE_*_DIR variables
 #-----------------------------------------------------------------------------
 if(Slicer_BUILD_CLI_SUPPORT)
-  set(Slicer_INSTALL_CLIMODULES_BIN_DIR "${Slicer_INSTALL_ROOT}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_BIN_DIR}")
-  set(Slicer_INSTALL_CLIMODULES_LIB_DIR "${Slicer_INSTALL_ROOT}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_LIB_DIR}")
-  set(Slicer_INSTALL_CLIMODULES_SHARE_DIR "${Slicer_INSTALL_ROOT}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_SHARE_DIR}")
+  set(Slicer_INSTALL_CLIMODULES_BIN_DIR "${Slicer_HOME}${Slicer_BUNDLE_EXTENSIONS_LOCATION}${Slicer_CLIMODULES_BIN_DIR}")
```
Comment (jcfr) [inline CMake/UseSlicer.cmake.in]: To review more carefully

File: Libs/ITKFactoryRegistration/CMakeLists.txt [hunk 2]
```diff
@@ -68,7 +68,7 @@ configure_file(
 # Install headers
 # --------------------------------------------------------------------------
 if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
+  set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ${Slicer_INSTALL_NO_DEVELOPMENT})
```
Comment (jcfr) [inline Libs/ITKFactoryRegistration/CMakeLists.txt]: This should already be set in https://github.com/Slicer/Slicer/blob/59da0db16500d7fbf8b1288748e59c28fad6bab0/Libs/CMakeLists.txt#L94

File: Libs/MRML/Widgets/CMakeLists.txt [hunk 3]
```diff
@@ -445,13 +445,18 @@ include_directories(${include_dirs})
...
+  set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ${Slicer_INSTALL_NO_DEVELOPMENT})
 endif()
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers}
     DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+
+  install(FILES
+    ${MRMLWidgets_UI_CXX}
```
Comment (jcfr) [inline Libs/MRML/Widgets/CMakeLists.txt]: We should revisit this.

File: Resources/CMakeLists.txt [hunk 4]
```diff
@@ -14,19 +14,19 @@ if(NOT Slicer_INSTALL_NO_DEVELOPMENT)
     )
 
   file(GLOB PNGFILES RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "*.png")
-  foreach(pngfile ${PNGFILES})
```
Comment (jcfr) [inline Resources/CMakeLists.txt]: Probably need to keep some version of this.

**Pull request closed without merging**